File: 백준/Silver/10828. 스택/스택.py
import sys

# n = int(sys.stdin.readline())

n = int(input())

stack = [] 

for i in range(n) :
  data = sys.stdin.readline().split()
  if(data[0] == "push") :
    stack.append(data[1])
  if(data[0] == "pop") :
    if len(stack) < 1 :
      print("-1")
    else :
      print(stack.pop())
  elif(data[0] == "size") :
      print(len(stack))
  elif(data[0] == "empty") :
    if len(stack) < 1 :
      print("1")
    else :
      print("0")
  elif(data[0] == "top") :
    if len(stack) < 1 :
      print("-1")
    else :
      print(stack[-1])
    

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);
	
	stack<int> S;
	int N;
	cin >> N;

	for (int i = 0; i < N; i++) {
		string str;
		cin >> str;
		if (str == "push") {
			int X;
			cin >> X;
			S.push(X);
		}
		else if (str == "top") {
			if (S.empty()) cout << -1 << '\n';
			else cout << S.top() << '\n';
		}
		else if (str == "size") cout << S.size() << '\n';
		else if (str == "empty") cout << S.empty() << '\n';
		else {
			if (S.empty()) cout << -1 << '\n';
			else {
				cout << S.top() << '\n';
				S.pop();
			}
		}
	}

	return 0;
}